### PR TITLE
[SPARK-25093][SQL] Avoid recompiling regexp for comments multiple times

### PR DESCRIPTION
**Pull request by mgaido91**
## What changes were proposed in this pull request?

The PR moves the compilation of the regexp for code formatting outside the method which is called for each code block when splitting expressions, in order to avoid recompiling the regexp every time.

Credit should be given to Izek Greenfield.

## How was this patch tested?

existing UTs

### CODE REVIEW

**Comment by mgaido91** [comment]
cc @cloud-fan @gatorsmile @kiszk

**Comment by mgaido91** [comment]
thanks for the comment @kiszk , I am doing it!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala [hunk 1]
```diff
@@ -91,10 +94,7 @@ object CodeFormatter {
   }
 
   def stripExtraNewLinesAndComments(input: String): String = {
-    val commentReg =
-      ("""([ |\t]*?\/\*[\s|\S]*?\*\/[ |\t]*?)|""" +    // strip /*comment*/
-       """([ |\t]*?\/\/[\s\S]*?\n)""").r               // strip //comment
-    val codeWithoutComment = commentReg.replaceAllIn(input, "")
+    val codeWithoutComment = commentRegexp.replaceAllIn(input, "")
     codeWithoutComment.replaceAll("""\n\s*\n""", "\n") // strip ExtraNewLines
```
Comment (igreenfield) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeFormatter.scala]: this line also compile regex and could be replaced!

**Review by igreenfield — COMMENTED** [review]
there is still one place the compile Regex again and again

**Comment by mgaido91** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**